resources/chef_ingredient.rb

```diff
@@ -0,0 +1,128 @@
+#
+# Author:: Joshua Timberman <joshua@chef.io>
+# Author:: Patrick Wright <patrick@chef.io>
+# Copyright (c) 2015-2017, Chef Software, Inc. <legal@chef.io>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+resource_name :chef_ingredient
+
+default_action :install
+
+property :product_name, kind_of: String, name_attribute: true
+property :config, kind_of: String
+
+# Determine what version to install from which channel
+property :version, kind_of: [String, Symbol], default: :latest
+property :channel, kind_of: Symbol, default: :stable, equal_to: [:stable, :current, :unstable]
+
+# Install package from local file
+property :package_source, kind_of: String
+
+# Set the *-ctl command to use when doing reconfigure
+property :ctl_command, kind_of: String
+
+# Package resources used on rhel and debian platforms
+property :options, kind_of: String
+property :timeout, kind_of: [Integer, String, NilClass]
+
+# Accept the license when applicable
+property :accept_license, kind_of: [TrueClass, FalseClass], default: false
+
+# Enable selecting packages built for earlier versions in
+# platforms that are not yet officially added to Chef support matrix
+# TODO: *this should not be defaulted. This should be handled by mixlib-install by default*
+property :platform_version_compatibility_mode, kind_of: [TrueClass, FalseClass], default: false
+
+# Configure specific platform package
+property :platform, kind_of: String
+property :platform_version, kind_of: String
+property :architecture, kind_of: String
+
+platform_family = node['platform_family']
```

Comment:
Why can't the `node['platform_family']` be placed down in the `action_class` block?